examples/stream_example.py

```diff
@@ -0,0 +1,70 @@
+import asyncio
+import sys
+from os import getenv
+
+from dotenv import load_dotenv
+from pydantic import BaseModel
+
+from examples.utils import make_blue, make_green, make_yellow
+from gpt_json import GPTJSON, GPTMessage, GPTMessageRole
+from gpt_json.models import GPTModelVersion
+from gpt_json.types_streaming import StreamEventEnum
+
+load_dotenv()
+API_KEY = getenv("OPENAI_API_KEY")
+
+SYSTEM_PROMPT = """
+You are a phenomenal and stubborn math tutor. Given the following problem, determine the best way to respond to the student by:
+1) Describing what the student does and does not understand about the problem in {student_model_key}
+2) Responding to the student's question in a way that does not give away the answer but helps them understand the problem better
+
+Problem: x^2 + 3 = 12 
+
+Respond with the following JSON schema:
+
+{json_schema}
+"""
+
+class TutorSchema(BaseModel):
+    student_model: str
+    tutor_response: str
+
+gpt_json = GPTJSON[TutorSchema](API_KEY, model=GPTModelVersion.GPT_4)
+
+async def main():
+    problem = "x^2 + 3 = 12"
+    print(make_blue("\nProblem:"), problem)
```

Comment:
I like the color-based formatting, but for right now I think let's keep the example directory just logic only and minimize additional code imports. If we want to extend the coverage in the future to something a bit more polished, I'd go in the direction of:
- Importing a `click` dependency
- Rewriting some of the entrypoints to these python files for better discoverability
- Using the `secho(fg="color")` function to do all the logging.